src/main/java/com/minecolonies/coremod/research/GlobalResearchTree.java

```diff
@@ -65,9 +139,63 @@ public void addResearch(final String branch, final IGlobalResearch research)
         {
             return Collections.emptyList();
         }
-        return researchTree.get(branch).values().stream().filter(research -> research.getParent().isEmpty()).map(IGlobalResearch::getId).collect(Collectors.toList());
+        return researchTree.get(branch).values().stream().filter(research -> research.getParent().isEmpty())
+                 .sorted(Comparator.comparing(IGlobalResearch::getResourceLocation))
+                 .map(IGlobalResearch::getId).collect(Collectors.toList());
+    }
+
+    @Override
+    public void reset()
```

Comment:
I still don't fully understand what and why it does this.

Reply:
Data packs are per-world, not per-Minecraft instance, while the Global Research Tree is per-Minecraft instance.  If someone logs into World A, which has several extra researches in its data pack, then into World B, which only has the defaults, the tree needs to be reset lest the client see the extra researches on a world that doesn’t have them.  

This gets worse for multiplayer.

I don’t think it’s a common failure mode, but it would be especially frustrating to diagnose due to its specific nature.

Reply:
Ahh, makes sense, okay